File: Module II/13. Databases/EXAM - DB/01. Code first/Superheroes/Superheroes.Models/Planet.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Superheroes.Models
{
    public class Planet
    {
        private ICollection<Country> countries;

        private ICollection<Fraction> fractions;

        public Planet()
        {
            this.countries = new HashSet<Country>();
            this.fractions = new HashSet<Fraction>();
        }

        public int Id { get; set; }

        [Index(IsUnique = true)]
        [MaxLength(30), MinLength(2)]
        [Required]
        public string Name { get; set; }

        public virtual ICollection<Country> Countries
        {
            get
            {
                return this.countries;
            }
            set
            {
                this.countries = value;
            }
        }

        public virtual ICollection<Fraction> Fractions
        {
            get
            {
                return this.fractions;
            }
            set
            {
                this.fractions = value;
            }
        }
    }
}

File: Module III/16.Node.Js/workshop-superheroes/superheroes/server/models/planet.js
"use strict";

const mongoose = require("mongoose");

const Schema = mongoose.Schema;

let planetSchema = new Schema({
    name: {
        type: String,
        required: true,
        minlength: 2, 
        maxlength: 30,
        unique: true
    },
    superheroes: [String],
    fractions: [String]
});

const Planet = mongoose.model("planet", planetSchema);
module.exports = Planet;